cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+ *---------------------------------------------------------------------------**/
+class strings_column_handler
+{
+ public:
+  ~strings_column_handler() = default;
+
+  strings_column_handler( column_view strings_column,
+                          rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource() );
+  //strings_column_handler( const column_view&& strings_column );
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of strings in the column
+   *---------------------------------------------------------------------------**/
+  size_type size() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal parent string column
+   *---------------------------------------------------------------------------**/
+  column_view parent_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal column of offsets
+   *---------------------------------------------------------------------------**/
+  column_view offsets_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the internal column of chars
+   *---------------------------------------------------------------------------**/
+  column_view chars_column() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal null mask memory
+   *---------------------------------------------------------------------------**/
+  const bitmask_type* null_mask() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of nulls in this column
+   *---------------------------------------------------------------------------**/
+  size_type null_count() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the registered memory resource
+   *---------------------------------------------------------------------------**/
+  rmm::mr::device_memory_resource* memory_resource() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the strings to stdout.
+   *
+   * @param start Index of first string to print.
+   * @param end Index of last string to print. Specify -1 for all strings.
+   * @param max_width Maximum number of characters to print per string.
+   *        Specify -1 to print all characters.
+   * @param delimiter The chars to print between each string.
+   *        Default is new-line character.
+   *---------------------------------------------------------------------------**/
+  void print( size_type start=0, size_type end=-1,
+              size_type max_width=-1, const char* delimiter = "\n" ) const;
+
+  // sort types can be combined
+  enum sort_type {
```

Comment:
Do these need to be in the class? Consider instead just putting the `enum` definition in the `cudf::strings` namespace.